### PR TITLE
T4generator line endings independence

### PR DESCRIPTION
**Pull request by pkar70**
GitHub Issue (If applicable): #

## PR Type

What kind of change does this PR introduce?
- Build or CI related changes

## What is the current behavior?
Sometimes we can get errors like
`Generating mixins .\Mixins\Android\FrameworkElementMixins.tt(1,1): error CS1519: Compiling transformation: Invalid token 'this' in class, struct, or interface member declaration`

## What is the new behavior?
No such errors.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [X] Tested code with current [supported SDKs](https://github.com/unoplatform/uno/blob/master/README.md#uno-features)
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] [Wasm UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md#running-the-webassembly-ui-tests-snapshots) are not showing unexpected any differences. Validate PR `Screenshots Compare Test Run` results.
- [X] Contains **NO** breaking changes
- [ ] Updated the [Release Notes](https://github.com/unoplatform/uno/tree/master/doc/ReleaseNotes)
- [ ] Associated with an issue (GitHub or internal)

## Other information
When I get new clone of Uno, my ability to build is vanished. I complain on Lobby, and Jérôme Laban says, "In most cases it's related to the .tt files having incorrect line endings", and "this is related to your default git configuration".
So, this small commit makes build independent of line endings (and one's default git configuration).

Internal Issue (If applicable):

### CODE REVIEW

**Comment by jeromelaban** [comment]
Thanks! Unfortunately, this solution does not work when building on macOS. There's probably a need to use the proper line ending based on the current platform.

**Comment by pkar70** [comment]
Seems strange that Microsoft.VisualStudio.TextTemplating is so 'fragile', than line ending could cause compile errors. I don't have macOS, so I cannot test it on Mac...
Are _ANDROID__ etc. flags accessible while t4generator is compiled? And what is proper line ending on mac? Classic mac, 0x0D, or newer mac, 0x0A?

And what means build error in this simple PR: `The nuget command failed with exit code(1) and error(Response status code does not indicate success: 500 (Internal Server Error - VS800075: The project with id 'vstfs:///Classification/TeamProject/1dd81cbd-cb35-41de-a570-b0df3571a196' does not exist, or you do not have permission to access it. (DevOps Activity ID: 55C7C3FE-6EBE-4015-9B34-2D6870CE734F)).)`

**Comment by jeromelaban** [comment]
For the line ending, using `Environment.NewLine` seems the most appropriate, let's try with that.

As for the not authorized error, this is a fork based PR error, I'll be fixing that in a separate PR.

**Comment by pkar70** [comment]
So, now it uses Environment.NewLine :)

File: src/T4Generator/Program.cs [hunk 1]
```diff
@@ -359,4 +362,26 @@ static void ProcessTemplate(string[] args)
             }
         }
     }
+
+	static class Tools
```
Comment (jeromelaban) [inline src/T4Generator/Program.cs]: That will do! Can you move the class to another file, and call it `StringHelper` maybe ?

Comment (pkar70) [reply src/T4Generator/Program.cs]: Which namespace? Or, class Tools, without namespace?

Comment (jeromelaban) [reply src/T4Generator/Program.cs]: Same namespace, another file.

**Comment by pkar70** [comment]
So, file StringHelper.cs, with one function ConvertLineEndings, in namespace CustomHost and class Tools?

**Comment by jeromelaban** [comment]
The file name and namespace does not matter much, it just has to not be in the same file.

**Comment by jeromelaban** [comment]
If you can, can you squash your history ? There are commits that are only merges.

**Comment by pkar70** [comment]
Sorry, I don't want to use squash - I tried to, but in effect one file was treated as new (before squash, it was only changed).

**Pull request merged**